File: 1460-number-of-substrings-containing-all-three-characters/1460-number-of-substrings-containing-all-three-characters.go
func numberOfSubstrings(s string) int {
    count, lastSeen := 0, []int{-1, -1, -1}
    for i := 0; i < len(s); i++ {
        lastSeen[s[i] - 'a'] = i
        count += 1 + min(lastSeen[0], lastSeen[1], lastSeen[2])
    }
    return count
}

File: 1460-number-of-substrings-containing-all-three-characters/1460-number-of-substrings-containing-all-three-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var numberOfSubstrings = function(s) {
    let count = 0, aLastSeen = -1, bLastSeen = -1, cLastSeen = -1;
    for (let i = 0; i < s.length; i++) {
        if (s[i] === "a") {
            aLastSeen = i;
        } else if (s[i] === "b") {
            bLastSeen = i;
        } else {
            cLastSeen = i;
        }
        count += 1 + Math.min(aLastSeen, bLastSeen, cLastSeen);
    }
    return count;
};